File: core/client_registry.py
import json
from pathlib import Path

CLIENTS_DIR = Path(__file__).parent.parent / "clients"


class ClientRegistry:
    def list_clients(self) -> list:
        clients = []
        for f in sorted(CLIENTS_DIR.glob("*.json")):
            try:
                with open(f, encoding="utf-8") as fp:
                    data = json.load(fp)
                    clients.append({
                        "id": data["client_id"],
                        "name": data["client_name"]
                    })
            except (json.JSONDecodeError, KeyError):
                continue
        return clients

    def load_client(self, client_id: str) -> dict:
        filepath = CLIENTS_DIR / f"{client_id}.json"
        if not filepath.exists():
            raise FileNotFoundError(f"No guidelines file found for client: {client_id}")
        with open(filepath, encoding="utf-8") as f:
            return json.load(f)

File: server/modules/gbpQc/clientRegistry.js
// Reads per-client GBP brand-guideline JSON files from clients/ — pure data,
// no persistence layer needed since new clients are onboarded by hand-adding
// a JSON file here, exactly as in the original standalone tool this was
// ported from.
const fs = require('fs');
const path = require('path');

const CLIENTS_DIR = path.join(__dirname, 'clients');

function listClients() {
  const clients = [];
  for (const file of fs.readdirSync(CLIENTS_DIR).sort()) {
    if (!file.endsWith('.json')) continue;
    try {
      const data = JSON.parse(fs.readFileSync(path.join(CLIENTS_DIR, file), 'utf8'));
      clients.push({ id: data.client_id, name: data.client_name });
    } catch {
      continue;
    }
  }
  return clients;
}

function loadClient(clientId) {
  const filepath = path.join(CLIENTS_DIR, `${clientId}.json`);
  if (!fs.existsSync(filepath)) {
    throw new Error(`No guidelines file found for client: ${clientId}`);
  }
  return JSON.parse(fs.readFileSync(filepath, 'utf8'));
}

module.exports = { listClients, loadClient };
